### PR TITLE
Added support for path_aliases.

### PR DESCRIPTION
**Pull request by jocelyn**
Refactored CMS_MODULE.router (..): WSF_ROUTER design,
  to create only one router object of type CMS_ROUTER.
Factorized sql builder initialization (to work for sqlite and mysql storage builders).
Code cleaning.

### CODE REVIEW

File: library/persistence/implementation/store/cms_storage_store_sql_builder.e [hunk 1]
```diff
@@ -0,0 +1,84 @@
+note
+	description: "[
+			Objects that ...
```
Comment (jvelilla) [inline library/persistence/implementation/store/cms_storage_store_sql_builder.e]: Missing description

File: src/persistence/cms_storage_null.e [hunk 2]
```diff
@@ -40,74 +45,23 @@ feature -- Status report
...
+	unset_path_alias (a_source: READABLE_STRING_8; a_alias: READABLE_STRING_8)
+			-- Unalias `a_source' from `a_alias'.
 		do
-			a_user.set_id (1)
 		end
 
-	update_user (a_user: CMS_USER)
-			-- Update user `a_user'.
+	resolved_path (a_alias: READABLE_STRING_8): detachable READABLE_STRING_8
+			-- Source path for alias `a_alias'.
```
Comment (jvelilla) [inline src/persistence/cms_storage_null.e]: Can you describe what's the function of this feature?

File: src/persistence/core/cms_core_storage_sql_i.e [hunk 3]
```diff
@@ -18,6 +18,73 @@ inherit
 
 	SHARED_LOGGER
 
+feature -- URL aliases
+
+	set_path_alias (a_source: READABLE_STRING_8; a_alias: READABLE_STRING_8)
+		local
```
Comment (jvelilla) [inline src/persistence/core/cms_core_storage_sql_i.e]: Missing description. <Precursor>

File: src/persistence/core/cms_core_storage_sql_i.e [hunk 4]
```diff
@@ -18,6 +18,73 @@ inherit
...
+				else
+					error_handler.add_custom_error (0, "alias exists", "Path alias %"" + a_alias + "%" already exists!")
+				end
+			else
+				sql_change (sql_select_insert_path_alias, l_parameters)
+			end
+		end
+
+	unset_path_alias (a_source: READABLE_STRING_8; a_alias: READABLE_STRING_8)
+		local
```
Comment (jvelilla) [inline src/persistence/core/cms_core_storage_sql_i.e]: Missign description Precurso

File: src/persistence/core/cms_core_storage_sql_i.e [hunk 5]
```diff
@@ -18,6 +18,73 @@ inherit
...
+		end
+
+	sql_select_path_alias: STRING = "SELECT source FROM path_aliases WHERE alias=:alias ;"
+			-- SQL select path aliases.
+
+	sql_select_insert_path_alias: STRING = "INSERT INTO path_aliases (source, alias) VALUES (:source, :alias);"
+			-- SQL insert path alias.			
+
+	sql_delete_path_alias: STRING = "DELETE FROM path_aliases WHERE alias=:alias;"
+				-- SQL delete path alias
```
Comment (jvelilla) [inline src/persistence/core/cms_core_storage_sql_i.e]: Wrong identation

**Comment by jocelyn** [comment]
Applied comments from Javier.

**Comment by jocelyn** [comment]
merge manually

**Pull request closed without merging**